### DISCUSSION TITLE
Multi-tenancy support for Prometheus

### DISCUSSION THREAD

**Discussion by venkatbvc** [grafana/grafana#67020] [Data Sources]
We have the request that metrics from the specified tenant in Prometheus shall be accessible by the users belonging to that tenant on Grafana.
On the side of Grafana, it supports Organization concept which can be used for purpose of tenancy. One Organization will map to one tenancy.
I think we need following enhancements:

• In Org structure, add two fields (Tenantlabel and Tenantvalue)
• Add proxy handling when invoking api/v1/query and api/v1/queryrange against Prometheus, the matching label '$Tenantlabel=$Tenantvalue' will be added into query parameter. The $Tenantlable and $Tenantvalue are getting from Org table via OrgID.
• Grafana frontend and restapi will be enhanced also.

This request is same as https://github.com/grafana/grafana/issues/15519, but we could not contribute due to other work items.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82725. Feel free to continue the discussion around this there. Thank you!
